src/applications/disability-benefits/all-claims/components/AutoComplete.jsx

```diff
@@ -0,0 +1,177 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { VaTextInput } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import debounce from 'lodash/debounce';
+import { fullStringSimilaritySearch } from 'platform/forms-system/src/js/utilities/addDisabilitiesStringSearch';
+
+const INSTRUCTIONS =
+  'When autocomplete results are available use up and down arrows to review and enter to select. Touch device users, explore by touch or with swipe gestures.';
+
+const createFreeTextItem = val => `Enter your condition as "${val}"`;
+
+const AutoComplete = ({
+  availableResults,
+  debounceDelay,
+  formData,
+  id,
+  label,
+  onChange,
+}) => {
+  const [value, setValue] = useState(formData);
+  const [results, setResults] = useState([]);
+  const [activeIndex, setActiveIndex] = useState(null);
+  const [ariaLiveText, setAriaLiveText] = useState('');
+
+  const inputRef = useRef(null);
+  const resultsRef = useRef([]);
+
+  // Delays screen reader result count reading to avoid interruption by input content reading
+  const debouncedSetAriaLiveText = useRef(
+    debounce((resultCount, freeTextResult) => {
+      const makePlural = resultCount > 1 ? 's' : '';
+
+      setAriaLiveText(
+        `${resultCount} result${makePlural}. ${freeTextResult}, (1 of ${resultCount})`,
+      );
+    }, 700),
+  ).current;
+
+  const debouncedSearch = useRef(
+    debounce(async inputValue => {
+      const freeTextResult = createFreeTextItem(inputValue);
+      const searchResults = fullStringSimilaritySearch(
+        inputValue,
+        availableResults,
+      );
+      const updatedResults = [freeTextResult, ...searchResults];
+      setResults(updatedResults);
+      setActiveIndex(0);
+
+      debouncedSetAriaLiveText(updatedResults.length, freeTextResult);
+    }, debounceDelay),
+  ).current;
+
+  const closeList = () => {
+    debouncedSearch.cancel();
+    debouncedSetAriaLiveText.cancel();
+    setResults([]);
+    setActiveIndex(null);
+  };
+
+  const handleInputChange = inputValue => {
+    setValue(inputValue);
+    onChange(inputValue);
+
+    if (!inputValue) {
+      closeList();
+      setAriaLiveText('Input is empty. Please enter a condition.');
+      return;
+    }
+
+    debouncedSearch(inputValue);
+  };
+
+  const scrollIntoView = index => {
```

Comment:
Just FYI no action needed: On my other team I'm doing some work with the scrolling behavior of the app not honoring `prefers-reduced-motion`, so when that ticket is done we should remember to come back and check what this combobox experience is like for someone who has that accessibility setting turned on https://github.com/department-of-veterans-affairs/va.gov-team/issues/95517